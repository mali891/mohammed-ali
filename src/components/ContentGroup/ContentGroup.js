import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { styles } from './ContentGroup.style'

const ContentGroup = React.memo(({ children, className, noMargin, ...other }) => {
  const dynamicClassNames = classNames([className], {
    'c-content-group': true,
    'c-content-group--no-margin': noMargin
  })

  return (
    <section css={styles} className={dynamicClassNames} {...other}>
      {children}
    </section>
  )
})

ContentGroup.defaultProps = {
  noMargin: false
}

ContentGroup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  noMargin: PropTypes.bool
}

export default ContentGroup
