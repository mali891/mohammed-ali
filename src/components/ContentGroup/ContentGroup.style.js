import { css } from '@emotion/core'
import { SPACING } from '../../styles/settings'

export const styles = css`
  display: block;
  width: 100%;
  margin-bottom: ${SPACING.XL};
`
