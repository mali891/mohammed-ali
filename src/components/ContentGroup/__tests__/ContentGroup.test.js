import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import ContentGroup from '../ContentGroup'

describe('ContentGroup', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<ContentGroup />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given a noMargin prop', () => {
    const { asFragment, container } = render(<ContentGroup noMargin />)

    expect(container.querySelector('.c-content-group--no-margin')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
