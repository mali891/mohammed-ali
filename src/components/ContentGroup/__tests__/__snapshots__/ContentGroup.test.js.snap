// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentGroup renders correctly 1`] = `
<DocumentFragment>
  <section
    class="c-content-group"
    css="[object Object]"
  />
</DocumentFragment>
`;

exports[`ContentGroup renders correctly given a noMargin prop 1`] = `
<DocumentFragment>
  <section
    class="c-content-group c-content-group--no-margin"
    css="[object Object]"
  />
</DocumentFragment>
`;
